internal/events/event.go

```diff
@@ -176,85 +177,67 @@ func (e *Events) V2AddMetricsEvent(ctx context.Context, clusterID *strfmt.UUID,
+			db = db.Where("cluster_id = ?", clusterID.String())
 		}
-	}
-	if hostID != nil {
-		host, err := common.GetHostFromDBbyHostId(e.db, *hostID)
-		if err != nil {
-			if errors.Is(err, gorm.ErrRecordNotFound) {
-				return &gorm.DB{Error: errors.Wrapf(gorm.ErrRecordNotFound, "host %s not found.", hostID.String())}
-			}
-			return &gorm.DB{Error: err}
+		if infraEnvID != nil {
+			db = db.Where("infra_env_id = ?", infraEnvID.String())
 		}
-		if user != "" && !identity.IsAdmin(ctx) { // A case where there is a non-admin user in context, the service should only present resources matching to it.
-			if host != nil && host.UserName == user {
-				whereCondition = append(whereCondition, fmt.Sprintf("host_id = '%s'", hostID.String()))
-			} else {
-				e.log.Errorf(
-					"user %s is not authorized to query events for host %s: host owned by another user.",
-					user, hostID.String())
-				return &gorm.DB{Error: errors.Wrapf(gorm.ErrRecordNotFound, "host %s not found.", hostID.String())}
-			}
-		} else { // A case where there is no user in context, simply filter by resource id
-			whereCondition = append(whereCondition, fmt.Sprintf("host_id = '%s'", hostID.String()))
+		if hostID != nil {
+			db = db.Where("host_id = ?", hostID.String())
 		}
+		return db
 	}
-	if infraEnvID != nil {
-		infraEnv, err := common.GetInfraEnvFromDB(e.db, *infraEnvID)
-		if err != nil {
-			if errors.Is(err, gorm.ErrRecordNotFound) {
-				return &gorm.DB{Error: errors.Wrapf(gorm.ErrRecordNotFound, "infra_env %s not found.", infraEnvID.String())}
-			}
-			return &gorm.DB{Error: err}
-		}
-		if user != "" && !identity.IsAdmin(ctx) { // A case where there is a non-admin user in context, the service should only present resources matching to it.
-			if infraEnv.UserName == user {
-				whereCondition = append(whereCondition, fmt.Sprintf("infra_env_id = '%s'", infraEnvID.String()))
-			} else {
-				e.log.Errorf(
-					"user %s is not authorized to query events for infra_env %s: infra_env owned by another user.",
-					user, infraEnvID.String())
-				return &gorm.DB{Error: errors.Wrapf(gorm.ErrRecordNotFound, "infra_env %s not found.", infraEnvID.String())}
-			}
-		} else { // A case where there is no user in context, simply filter by resource id
-			whereCondition = append(whereCondition, fmt.Sprintf("infra_env_id = '%s'", infraEnvID.String()))
-		}
+
+	//prepare the common parts of the query
+	db := e.db.Debug().Order("event_time").Where("category IN (?)", selectedCategories)
+	if e.authz != nil {
+		db = e.authz.OwnedBy(ctx, db)
 	}
-	if len(whereCondition) == 0 {
-		queryErr := errors.Wrap(gorm.ErrInvalidTransaction, "events query with no filters is not allowed.")
-		e.log.Error(queryErr)
-		return &gorm.DB{Error: queryErr}
+
+	var result *gorm.DB
+
+	//retrieveing all events can be done only by admins. This is done to restrict data
+	//intensive queries by common users
+	if clusterID == nil && infraEnvID == nil && hostID == nil && identity.IsAdmin(ctx) {
```

Comment:
maybe worth having an helper func for 'clusterID == nil && infraEnvID == nil && hostID == nil' - something like isAllEventsRequested.

Reply:
OK. I will add helpers for all the cases

Reply:
Should we keep a debug log in place for this case?